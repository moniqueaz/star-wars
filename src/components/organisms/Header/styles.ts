import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${( { theme } ) => css`
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background-image: linear-gradient(to bottom, ${theme.colors.primary}, ${theme.colors.mainBg});
    position: relative;
    z-index: 1;
    display: flex;
    justify-content: center;
    align-items: center;
    height: ${theme.height.header.mobile};
    @media(min-width: 1024px){
      height: ${theme.height.header.desk};
    }

    &:before{
      content: '';
      background-image: linear-gradient(to right, transparent, #fff, transparent);
      height: 2px;
      width: 100%;
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      opacity: .5;
      z-index: 0;
      animation: show 1s linear;
      
      @media(min-width: 1024px){
        height: 3px;
      }
    }

    @keyframes show {
      from {width: 0};
      to {width: 100%};
    }
  `}
`;
