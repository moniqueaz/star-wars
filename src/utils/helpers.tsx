export const fetcher = (url: string) => fetch(url).then(res => res.json());

export const getRandomInteger = (min: number, max: number) => {
  return Math.floor(Math.random() * (max - min)) + min;
};
