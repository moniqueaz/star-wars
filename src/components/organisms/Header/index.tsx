import * as S from './styles';
import Logo from 'components/atoms/Logo';

const Header = () => (
  <S.Wrapper>
    <Logo />
  </S.Wrapper>
);

export default Header;
