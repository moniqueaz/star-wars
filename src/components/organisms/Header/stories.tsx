import { Story, Meta } from '@storybook/react/types-6-0';
import Header from '.';

export default {
  title: 'Organisms/Header',
  component: Header,
} as Meta;

export const Default: Story = () => {
  return <Header />;
};
