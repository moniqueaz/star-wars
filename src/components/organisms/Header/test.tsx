import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import Header from '.';

jest.mock('components/atoms/Logo', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Logo"/>;
    },
  };
} );

describe('<Header />', () => {
  it('should render', () => {
    const { container } = renderWithTheme(<Header />);
    expect(screen.getByTestId('Mock Logo')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  } );
} );
