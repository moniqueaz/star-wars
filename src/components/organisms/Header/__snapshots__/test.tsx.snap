// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-image: linear-gradient(to bottom,#1A4173,#0C1D33);
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 4.8rem;
}

.c0:before {
  content: '';
  background-image: linear-gradient(to right,transparent,#fff,transparent);
  height: 2px;
  width: 100%;
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  opacity: .5;
  z-index: 0;
  -webkit-animation: show 1s linear;
  animation: show 1s linear;
}

@media (min-width:1024px) {
  .c0 {
    height: 14rem;
  }
}

@media (min-width:1024px) {
  .c0:before {
    height: 3px;
  }
}

<div
  class="c0"
>
  <div
    data-testid="Mock Logo"
  />
</div>
`;
